### ISSUE TITLE
kubectl should support [*] for strings in jsonpath

### ISSUE THREAD

**Issue by mengqiy** [kubernetes/kubernetes#39500]
Originally from kubernetes/charts#366

```
kubectl get secret mysecret -o jsonpath="{.data.password[*]}"
Error executing template: string is not array or slice. Printing more information for debugging the template:
	template was:
		{.data.password[*]}
	object given to jsonpath engine was:
		map[string]interface {}{"type":"Opaque", "kind":"Secret", "apiVersion":"v1", "metadata":map[string]interface {}{"name":"mysecret", "namespace":"default", "selfLink":"/api/v1/namespaces/default/secrets/mysecret", "uid":"87f682e9-d381-11e6-aa89-42010a800002", "resourceVersion":"2981914", "creationTimestamp":"2017-01-05T19:59:56Z"}, "data":map[string]interface {}{"password":"MWYyZDFlMmU2N2Rm", "username":"YWRtaW4="}}

error: error executing jsonpath "{.data.password[*]}": string is not array or slice
```

According to https://github.com/kubernetes/charts/issues/366#issuecomment-270776575, kubectl used to support `[*]` for strings in 1.4.
But kubectl now complains about using `[*]` for strings in 1.5.

@kubernetes/sig-cli-misc

**Comment by viglesiasce** [comment]
Over in https://github.com/kubernetes/charts/issues/366#issuecomment-274391763  @iamzhout posits that this is caused by Go 1.7 vs Go 1.6

**Comment by mengqiy** [comment]
Closing this since it has been resolved, see kubernetes/charts#366
